src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -1333,7 +1333,15 @@ trait Contexts { self: Analyzer =>
               val pre2 = companionModule.typeOfThis
               val found = pre2.decl(name).filter(qualifies)
               found match {
-                case NoSymbol => NoJavaMemberFound
+                case NoSymbol =>
+                  if (name.isTermName) {
+                    val found = pre2.decl(name.toTypeName).filter(qualifies)
+                    if (found.isJavaEnum)
+                      (pre2, found)
```

Comment:
I worry that returning a type member for a term name lookup could leak. Is that why you narrowed it to enums, to limit the scope?